integrations/gcp/gcp_core/helpers/ratelimiter/base.py

```diff
@@ -0,0 +1,178 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional, TYPE_CHECKING, final
+
+from aiolimiter import AsyncLimiter
+from google.cloud.cloudquotas_v1 import CloudQuotasAsyncClient, GetQuotaInfoRequest  # type: ignore
+from google.api_core.exceptions import GoogleAPICallError
+from loguru import logger
+from enum import Enum
+from port_ocean.context.ocean import ocean
+from gcp_core.cache import cache_coroutine_result
+from collections.abc import MutableSequence
+import asyncio
+
+
+_DEFAULT_RATE_LIMIT_TIME_PERIOD: float = 60.0
+_DEFAULT_RATE_LIMIT_QUOTA: int = int(
+    ocean.integration_config["search_all_resources_per_minute_quota"]
+)
+_PERCENTAGE_OF_QUOTA: float = 0.8
```

Comment:
```suggestion
_PERCENTAGE_OF_QUOTA: float = 0.9
```

Reply:
The closer we get to the quota, the higher our chances of getting close to the rate limit and also, further to the retry mechanism.
Ideally, we don't want to be approaching the retry mechanism a lot because at the moment its retry's an entire resource and not a page. 

I must also add, that though we experience little to no rate limits now from tests I've done, I do see randomly (rare but observed) that the rate limit does hit, good part is that in such scenario's the retry mechanism does set in to compensate. However, getting away from the quota value by 20-30% does reduce chances of this happening.